### ISSUE TITLE
Community adoption gates: external users, reproducibility challenges, and methods-paper readiness

### ISSUE THREAD

**Issue by sidhulyalkar** [sidhulyalkar/neurOS-v1#76]
## Why

Peer acceptance cannot be established by repository-internal CI alone. neurOS needs evidence that independent researchers can understand, install, run, challenge, extend, and cite the platform.

This issue complements #75 by turning community/scientific acceptance into measurable release gates.

### 1. Independent reproduction challenge
- [ ] publish one small frozen Neural System Qualification Benchmark v1 task with a one-command reproduction path;
- [ ] recruit at least 3 independent users who were not involved in implementation;
- [ ] require each user to start from public documentation and built wheels only;
- [ ] preserve all failures/issues from those attempts;
- [ ] publish a reproducibility report summarizing setup time, failures, fixes and final result hashes.

### 2. External model submission
- [ ] accept at least one model implemented entirely outside neurOS through the public adapter/protocol;
- [ ] run it under the same frozen authority as built-in baselines;
- [ ] ensure the submitter does not need to edit neurOS internals;
- [ ] preserve attribution/citation metadata for external methods.

### 3. External dataset/integration path
- [ ] reproduce the benchmark through upstream MNE/MOABB/Braindecode APIs rather than private copies;
- [ ] add one external plugin in its own repository using the maintained plugin kit;
- [ ] qualify one real device or real public-data integration without changing `neuros-core`.

### 4. Community contribution gate
- [ ] land at least one substantive external PR;
- [ ] demonstrate that contributor setup, tests and review guidance are sufficient without maintainer handholding;
- [ ] collect contributor feedback and convert recurring friction into contract/documentation tests.

### 5. Methods-paper readiness
- [ ] write a concise Statement of Need centered on neural-system qualification;
- [ ] write a State of the Field comparing MNE, Braindecode, MOABB, NWB/BIDS, BrainFlow/LSL, SpikeInterface, NeuroBench, Brain-Score and Meta NeuroAI;
- [ ] include explicit build-vs-contribute justification;
- [ ] describe software design tradeoffs, not merely package layout;
- [ ] include reproducible research-impact evidence rather than future aspirations;
- [ ] disclose AI-assisted development and the independent verification process transparently;
- [ ] create archived tagged release/DOI only once a scientifically coherent external-user release exists.

### 6. Public project signals
- [ ] enable an appropriate public discussion/support channel;
- [ ] publish a short external roadmap with `good first issue` / integration tasks;
- [ ] add benchmark result tables generated from machine-readable artifacts rather than manually curated claims;
- [ ] publish a release cadence and compatibility horizon external labs can plan against.

## Anti-goal

Do not optimize stars, package count, benchmark count or model count directly. The target is independent reproducibility and scholarly usefulness. One external lab relying on neurOS qualification for a real paper is worth more than a large amount of repository-local sophistication.

**Comment by sidhulyalkar** [comment]
## 2026-08-29 external-adoption audit

The implementation is now ahead of the public product surface. Current repository signals are still weak for discovery/adoption: 3 stars, 0 forks, no repository topics, no homepage, no Discussions, no GitHub Releases, an outdated repository description centered on streaming/classification, and GitHub currently identifies Jupyter Notebook as the primary language even though the maintained platform is Python.

The biggest adoption gap is therefore not another model or package. It is the path from **landing on GitHub -> understanding the unique job -> producing a useful evidence artifact -> bringing existing code into the referee -> contributing/reproducing independently**.

PR #103 is the first concrete productization slice:

- adds `neuros init` with a minimal project that validates through the production config resolver;
- qualifies the generated project from a clean built-wheel environment through validate/run/qualify/reproduce;
- rewrites the README around neurOS as the qualification/reproducibility layer rather than a package inventory;
- adds a first-class Bring Your Own Model path showing that sklearn/Braindecode/lab code does not need to adopt neurOS training code;
- adds structured issue forms for reproducible bugs, ecosystem integrations, and external method participation;
- marks research notebooks as documentation for GitHub Linguist so language stats better represent the implementation.

### Adoption gates I would now treat as release blockers

1. `neuros init` installed-wheel journey green on an exact PR/release head.
2. One tagged **Developer Preview** release with wheels, hashes, release manifest, and a clear supported package/install story.
3. A one-command frozen public NSQ benchmark slice that an outsider can reproduce without understanding the monorepo.
4. At least one external method submitted without modifying neurOS internals.
5. At least one independent reproduction from outside the implementation loop.
6. GitHub storefront cleanup: accurate description, topics, docs homepage, Discussions, release notes.
7. A small set of genuinely newcomer-sized issues after the public surfaces stabilize.

The target should not be stars by themselves. The stronger signal is that external researchers choose neurOS because it lets them make a neural-system claim with a clearer, harder-to-game evidence boundary.

**Comment by sidhulyalkar** [comment]
### 2026-08-29 adoption checkpoint

The public-product path has moved from roadmap to qualified `main` behavior.

Now merged:
- #106: one-command `neuros init` project on-ramp, first-10-minutes/BYOM docs, structured community forms, Developer Preview installed-wheel qualification. Exact product head `a4c67d2d...` passed 13/13 workflows before merge; main merge `ddf63444...`.
- #107: `neuros init --template nsq-method`, generating a normal sklearn external method that enters the production NSQ referee. Exact head `66081f45...` passed 6/6 workflows before merge; main merge `c4b2470a...`.

Fresh #107 adapter evidence was independently inspected: GitHub artifact ZIP digest `4ed1174e35bfca2f45995e39ad9a382c6c7446b9ef95402981f71af8a9552b2b` matched local recomputation; five calibration budgets produced five distinct run-contract identities and five distinct content-addressable external learned-state identities while preserving one immutable final-assessment identity. The synthetic numerical outputs remain explicitly non-interpretable.

The next adoption bottlenecks are therefore not another starter or model implementation. They are: protected `main`; coherent Developer Preview publishing authority/tagged release; GitHub description/topics/homepage/Discussions; a one-command frozen public real-data NSQ slice after its execution authority qualifies; and, most importantly, independent external reproduction/contribution.

Recommended public category remains: **qualification and reproducible execution for neural AI and BCI systems**.

**Comment by sidhulyalkar** [comment]
Concrete external-adoption entry points now exist:

- #114 `good first issue` / `help wanted`: an independent user runs the First 10 Minutes journey from a clean environment using only public docs and reports exact friction/hashes.
- #115 `good first issue` / `help wanted`: a contributor adds a small pure-scikit-learn external-style NSQ adapter example from the public `nsq-method` starter, using deterministic synthetic data and no promoted efficacy claim.

These are intentionally narrower than the larger adoption gates here. They provide honest first contributions that do not require understanding the internal scientific promotion history or modifying frozen Kumar2024 authority.

**Comment by sidhulyalkar** [comment]
## 2026-08-30 priority reset: external challenge is now a release-blocking engineering input

The repository has crossed a threshold where additional internal contract breadth has diminishing trust value relative to independent failure discovery.

As of the current frozen `main`, the three most useful external contribution tasks are now explicitly surfaced:

- #114 `good first issue` + `help wanted`: clean-environment First 10 Minutes reproduction;
- #115 `good first issue` + `help wanted`: external-style scikit-learn NSQ method integration;
- #116 `help wanted`: independently reimplement/reproduce the promoted Kumar2024 binding audit without importing neurOS's verifier.

These three tasks intentionally attack three different failure modes:

```text
#114  developer usability / undocumented assumptions
#115  API boundary / framework capture / external method friction
#116  cryptographic + scientific authority self-verification risk
```

### Revised developer-preview external gate

Before describing the first coordinated developer preview as externally validated, require at least:

1. one person outside the implementation loop completes #114 from public instructions;
2. one external-style method is run through NSQ without modifying neurOS internals, preferably by someone other than the maintainer;
3. one independent binding verifier reproduces or falsifies the frozen authority roots through separate code;
4. every failure encountered is preserved as an issue/test/documentation fix rather than silently coached around;
5. the repeated path is rerun after fixes by the outsider or a second person.

These are stronger signals than more repository-local green checks because they test whether the contracts are understandable and independently reconstructible.

### Product metrics worth tracking

Do not optimize stars as the primary metric. Track things that reflect whether neurOS removes real trust friction:

- median time from clean machine to first verified qualification artifact;
- number of undocumented/manual interventions required;
- time for an external model author to integrate without internal imports;
- percentage of frozen evidence artifacts independently verifiable from documented schemas;
- number and severity of discrepancies found by independent verification;
- percentage of benchmark methods/data paths using public adapters rather than privileged internal APIs;
- artifact reproduction success across supported Python/platform environments;
- later, number of external papers/labs that cite a neurOS evidence identity rather than only the package.

### Development policy until this gate gets traction

Unless work is required for the frozen Kumar2024 critical path, security/reproducibility, or removing duplicate authority, prefer **external adoption friction** over new internal subsystem breadth.

A new package/model/UI/cloud feature should lose priority to a reproducible outsider failure. This is not slowing development; it is moving development onto the interface where product truth is generated.
